datasync/chaindatafetcher/kafka/consumer.go

```diff
@@ -65,8 +68,30 @@ func newSegment(msg *sarama.ConsumerMessage) (*Segment, error) {
 		return nil, errors.New(nilConsumerMessageErrorMsg)
 	}
 
-	if len(msg.Headers) != MsgHeaderLength {
-		return nil, fmt.Errorf("%v [header length: %v]", wrongHeaderNumberErrorMsg, len(msg.Headers))
+	headerLen := len(msg.Headers)
+	if headerLen != MsgHeaderLength && headerLen != LegacyMsgHeaderLength {
+		return nil, fmt.Errorf("%v [header length: %v]", wrongHeaderNumberErrorMsg, headerLen)
+	}
+
+	version := ""
+	producerId := ""
+
+	if len(msg.Headers) == MsgHeaderLength {
+		keyVersion := string(msg.Headers[MsgHeaderVersion].Key)
```

Comment:
I may have misunderstood.
I think it have a dependency on the header record order.
Is it okay..?

Reply:
I agree with you. The order or the header is also checked previously as well, but it is too strict to check. 
However, let me update about this later.

Reply:
Thank you for your kind answer!